sysutils/opentofu/Portfile

```
@@ -0,0 +1,144 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           select 1.0
+
+name                opentofu
+categories          sysutils
+maintainers         {@graywolf wolfsden.cz:~+macports} openmaintainer
+license             MPL-2
+
+homepage            https://opentofu.org/
+
+installs_libs       no
+
+# Shamelessly inspired by terraform's Portfile
+proc opentofuBaseVersion {} {
+    global subport
+    return [lindex [split ${subport} "-"] 1]
+}
+
+proc opentofuVersion {} {
+    global patchNumber
+    return [opentofuBaseVersion].${patchNumber}
+}
+
+proc opentofuDistBase {} {
+    global name
+    return tofu_[opentofuVersion]_darwin
+}
+
+# *NOTE* Remember to update `latestVersion` on a version upgrade.
+set latestVersion       opentofu-1.8.8
+
+subport opentofu-1.8 {
+    set patchNumber 8
+    revision        0
+
+    checksums       [opentofuDistBase]_amd64.tar.gz \
+                    rmd160  574f8be74cd6a9d252ceb4172daa9b7b133f74be \
+                    sha256  9dd27362eb247abe181b6825ae72af37bc0b9e4dc74df2392dc681761fef520f \
+                    size    26008291 \
+                    [opentofuDistBase]_arm64.tar.gz \
+                    rmd160  7bf8f69f8527032dacbc44c0fbea24d98faf69e0 \
+                    sha256  dab912199ff7703c8ce8e7679dcac57a18f9bfc2f60a12454c82e74a3e24ca78 \
+                    size    24556716
+}
+
+subport opentofu-1.7 {
+    set patchNumber 7
+    revision        0
+
+    checksums       [opentofuDistBase]_amd64.tar.gz \
+                    rmd160  799695b039d6a9108bcd86553b3d580a66992e48 \
+                    sha256  94fdd023c38ee2f1765fec43eff59ed1a91b24e10a517b31de128eccc0b8fd24 \
+                    size    25922637 \
+                    [opentofuDistBase]_arm64.tar.gz \
+                    rmd160  cb6d471cdcc681a0cea84bb21e6706595d8c7fbe \
+                    sha256  7253fbb4fd2448e0480d6f3f567c569b108507ddfc6fc1c6c2e956eddaba11d7 \
+                    size    24471570
+}
+subport opentofu_select {}
+
+if {${subport} eq ${name}} {
+    PortGroup           obsolete 1.0
+
+    replaced_by         ${latestVersion}
+    version             1.8.8
+    revision            0
+
+} elseif {${subport} eq "opentofu_select"} {
+    version         0.0.0
+    revision        0
+    supported_archs noarch
+    platforms       any
+
+    description     Common files for selecting the default opentofu \
+                    version
+    long_description \
+        This port installs files that allow 'port select' \
+        to create links to the preferred default version of opentofu.
+    homepage        https://www.macports.org/
+
```

```suggestion

    conflicts       tenv

```

@graywolf Apologies for the two earlier duplicate notifications. Mistakes were made with those previous review comments.

A new port [`tenv`](https://ports.macports.org/port/tenv) was recently added. Since it installs a `tofu` binary in `${prefix}/bin`, it will conflict with this OpenTofu port. It currently already [conflicts with the Terraform port](https://github.com/macports/macports-ports/blob/67821ec5d8849f94e3562c8fd681ec6e58afaa2b/sysutils/tenv/Portfile#L17) since it also installs a `terraform` binary.